src/common/string_util.cc

```diff
@@ -276,6 +276,71 @@ std::pair<std::string, std::string> SplitGlob(std::string_view glob) {
   return {prefix, ""};
 }
 
+StatusOr<std::vector<std::string>> SplitArguments(std::string_view in) {
+  std::vector<std::string> arguments;
+  std::string current_string;
+
+  enum State { NORMAL, DOUBLE_QUOTED, SINGLE_QUOTED, ESCAPE } state = NORMAL;
+
+  State state_before_escape = NORMAL;
+  for (const char c : in) {
+    switch (state) {
+      case NORMAL:
+        if (std::isspace(c)) {
+          if (!current_string.empty()) {
+            arguments.emplace_back(std::move(current_string));
+            current_string.clear();
+          }
+          // skip spaces
+        } else if (c == '"' || c == '\'') {
+          state = c == '"' ? DOUBLE_QUOTED : SINGLE_QUOTED;
+        } else {
+          current_string.push_back(c);
+        }
+        break;
+      case DOUBLE_QUOTED:
+      case SINGLE_QUOTED:
+        if (c == '\\') {
+          state_before_escape = state;
+          state = ESCAPE;
+        } else if ((c == '"' && state == DOUBLE_QUOTED) || (c == '\'' && state == SINGLE_QUOTED)) {
+          state = NORMAL;
+        } else {
+          current_string.push_back(c);
+        }
+        break;
+      case ESCAPE:
+        if (c == '"' || c == '\'' || c == '\\') {
+          current_string.push_back(c);
+        } else if (c == 't') {
+          current_string.push_back('\t');
+        } else if (c == 'r') {
+          current_string.push_back('\r');
+        } else if (c == 'n') {
+          current_string.push_back('\n');
+        } else if (c == 'v') {
+          current_string.push_back('\v');
+        } else if (c == 'f') {
+          current_string.push_back('\f');
+        } else if (c == 'b') {
+          current_string.push_back('\b');
+        }
```

Comment:
The redis code is:

```c
switch(*p) {
                        case 'n': c = '\n'; break;
                        case 'r': c = '\r'; break;
                        case 't': c = '\t'; break;
                        case 'b': c = '\b'; break;
                        case 'a': c = '\a'; break;
                        default: c = *p; break;
                        }
```

Should we `push_back(c)` after checks above?

Reply:
what's the meaning of this comment, Redis also emplaces the escaped char: https://github.com/redis/redis/blob/b33a405bf16cfa147f1736a6a3cd64c2fd2b3f7d/src/sds.c#L1030

Reply:
I think the meaning is that, we need to consider cases like `\q`, where `q` is not a valid escape char.

Reply:
Seems here we just ignore it.

Reply:
I mean the final line:
```
 default: c = *p; break;
```

In our "ESCAPE" mode, this is absent?

Reply:
Oh, I see. Good point. It's now ignored, and we should continue to behave the same way with Redis. I will fix them when I have time later.